accounts-db/src/accounts_db.rs

```diff
@@ -6888,14 +6883,13 @@ impl AccountsDb {
 
         match store_to {
             StoreTo::Cache => {
-                let txn_iter: Box<dyn std::iter::Iterator<Item = &Option<&SanitizedTransaction>>> =
-                    match transactions {
-                        Some(transactions) => {
-                            assert_eq!(transactions.len(), accounts.len());
-                            Box::new(transactions.iter())
-                        }
-                        None => Box::new(std::iter::repeat(&None).take(accounts.len())),
-                    };
+                let txn_iter = match transactions {
+                    Some(transactions) => {
+                        assert_eq!(transactions.len(), accounts.len());
+                        transactions.iter().copied()
+                    }
+                    None => [].iter().copied(),
```

Comment:
behavior change here - instead of creating a box for different types of iterators, we either have an iterator over the txns vs an empty iterator.

At the next level down we just use `None` if the iterator has no next element.

Reply:
de664e8f5f8238d2793b980f735857a807c70ce1